platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/creation/rules/EventCreationRule.java

```diff
@@ -32,4 +35,13 @@ public interface EventCreationRule {
      * This method is called whenever an event is created.
      */
     void eventWasCreated();
+
+    /**
+     * If event creation is blocked by this rule, this method should return the status that the event creator should
+     * adopt.
+     *
+     * @return the status that the event creator should adopt if this rule blocks event creation
+     */
+    @NonNull
+    EventCreationStatus getEventCreationStatus();
```

Comment:
Have you considered replacing `isEventCreationPermitted` with a method that does the check, and returns a resulting `EventCreationStatus`? That seems to me like a cleaner API than setting an internal value when calling `isEventCreationPermitted`, and then fetching that value with `getEventCreationStatus`

Reply:
We discussed this on a call, the decision was to not change this API at this time.